pygen_src/riscv_instr_pkg.py

```diff
@@ -0,0 +1,1595 @@
+#The offset from the address of <main> - automatically populated by the PMP generation routine.
+
+else:
+    class pmp_cfg_reg_t(Enum):   # struct
+        l = randint(0,)
+	zero = auto()
+	a = randint(0,)     # pmp_addr_mode_t a;
+	x = randint(0,)
+	w = randint(0,)
+	r = randint(0,)
+	addr = auto()
+	offset = auto()
+
+
+
+def hart_prefix(hart = 0):
+    if NUM_HARTS <= 1:
+        return ""
+    else:
+        return print("hart")
+
+
+
+
+
+def get_label(label, hart = 0):
+    return {hart_prefix(hart), label}   
+
+
+
+
+class vtype_t(Enum):
+    ill = auto() 
+    reserved = auto()                    # bit [XLEN-2:7] reserved
+    vediv = auto()  
+    vsew = auto() 
+    vlmul = auto() 
+
+class vxrm_t(Enum):
+    RoundToNearestUp = auto()
+    RoundToNearestEven = auto()
+    RoundDown = auto()
+    RoundToOdd = auto()
+
+class b_ext_group_t(Enum):
+    ZBB = auto()
+    ZBS = auto()
+    ZBP = auto()
+    ZBE = auto()
+    ZBF = auto()
+    ZBC = auto()
+    ZBR = auto()
+    ZBM = auto()
+    ZBT = auto()
+    ZB_TMP = auto()
+    #for uncategorized instructions
+
+
+
+#`VECTOR_INCLUDE("riscv_instr_pkg_inc_variables.sv")   # to do.....
+
+
+
+# typedef bit [15:0] program_id_t;    #to do........
+
+
+
+# xSTATUS bit mask
+
+def MPRV_BIT_MASK():                # parameter bit[XLEN - 1: 0] MPRV_BIT_MASK = 'h1 << 17
+    return 0x1 << 17
+
+def SUM_BIT_MASK():                 # parameter bit [XLEN - 1 : 0] SUM_BIT_MASK  = 'h1 << 18
+    return 0x1 <<18
+
+def MPP_BIT_MASK():                 # parameter bit [XLEN - 1 : 0] MPP_BIT_MASK  = 'h3 << 11
+    return 0x3 << 11
+
+
+def IMM25_WIDTH():                  # parameter int IMM25_WIDTH = 25
+    return 25
+
+def IMM12_WIDTH():                  # parameter int IMM12_WIDTH = 12
+    return 12
+
+def INSTR_WIDTH():                  # parameter int INSTR_WIDTH = 32;
+    return 32
+
+def DATA_WIDTH():                   # parameter int DATA_WIDTH  = 32;
+    return 32
+
+
+#Parameters for output assembly program formatting
+
+def MAX_INSTR_STR_LEN():             # parameter int MAX_INSTR_STR_LEN = 11;
+    return 11
+
+def LABEL_STR_LEN():                 # parameter int LABEL_STR_LEN     = 18;
+    return 18
+
+
+#Parameter for program generation
+
+def MAX_CALLSTACK_DEPTH():           # parameter int MAX_CALLSTACK_DEPTH = 20;
+    return 20
+
+def MAX_SUB_PROGRAM_CNT():           # parameter int MAX_SUB_PROGRAM_CNT = 20;
+    return 20
+
+def MAX_CALL_PER_FUNC():             # parameter int MAX_CALL_PER_FUNC   = 5;
+    return 5
+
+def indent():                        # string indent = {LABEL_STR_LEN{" "}};
+    return {LABEL_STR_LEN{" "}}
+
+
+
+ 
+# Format the string to a fixed length
+
+def format_string(str, len=10):
+
+    formatted_str = auto()
+
+    formatted_str = {len{" "}}     # formatted_str = {len{" "}};
+
+    if (len < str.len()):
+        return str
+
+    formatted_str = {str, formatted_str.substr(0, len-str.len() - 1)}

```

Comment:
Please check this whole function it seems like you have copy-pasted directly from SV